state/stateenvirons/policy.go

```diff
@@ -23,14 +23,16 @@ import (
 // environStatePolicy implements state.Policy in
 // terms of environs.Environ and related types.
 type environStatePolicy struct {
-	st                   *state.State
-	cloudService         CloudService
-	credentialService    CredentialService
-	getEnviron           NewEnvironFunc
-	getBroker            NewCAASBrokerFunc
-	checkerMu            sync.Mutex
-	checker              deployChecker
-	storageServiceGetter storageServiceGetter
+	st                       *state.State
+	cloudService             CloudService
+	credentialService        CredentialService
+	modelConfigService       ModelConfigService
+	modelConfigServiceGetter modelServiceGetter
```

Comment:
Do we really need both of these?

Reply:
They are used by different parts of the policy. The getter is a fallback if the model service itself cannot be provided. This happens in some of the facades.

Reply:
From what I can see, the `modelConfigService` is never assigned a value, i.e. it is always nil. I think we could blow this away and just use the `modelConfigServiceGetter`.